R/documentation.R

```diff
@@ -192,6 +192,53 @@ NULL
 #' @name Button
 NULL
 
+#' FileUploadButton
+#'
+#' @description
+#' A Safari-compatible file upload button that combines Fluent UI styling with cross-browser file selection functionality. This component solves the issue where Safari blocks programmatic file input clicks by using React-based file input handling.
+#'
+#' For more details about Fluent UI buttons visit the [official docs](https://developer.microsoft.com/en-us/fluentui#/controls/web/Button).
+#' The R package cannot handle each and every case, so for advanced use cases
+#' you need to work using the original docs to achieve the desired result.
+#'
+#' @param inputId ID of the component.
+#' @param value Starting value.
+#' @param session Object passed as the `session` argument to Shiny server.
+#' @param ... Props to pass to the component.
+#' The allowed props are listed below in the \bold{Details} section.
+#'
+#' @section Best practices:
+#' ### Usage
+#' - Use FileUploadButton when you need Safari-compatible file uploads
+#' - Choose appropriate buttonType for visual hierarchy (primary for main actions)
+#' - Set meaningful accept attributes to filter file types
```

Comment:
Issue: the current phrasing is a bit odd
```suggestion
#' - Use `accept` prop to filter allowed file extensions
```